R/obic.R

```diff
@@ -574,66 +600,235 @@ obic_field_dt <- function(dt,output = 'all') {
+                    A_S_RT = dt$A_S_RT,
+                    A_N_PMN = dt$A_N_PMN,
+                    A_P_AL= dt$A_P_AL, 
+                    A_P_CC = dt$A_P_CC, 
+                    A_P_WA = dt$A_P_WA,
+                    A_CEC_CO = dt$A_CEC_CO,
+                    A_CA_CO_PO = dt$A_CA_CO_PO, 
+                    A_MG_CO_PO = dt$A_MG_CO_PO, 
+                    A_K_CO_PO = dt$A_K_CO_PO,
+                    A_K_CC = dt$A_K_CC,
+                    A_MG_CC = dt$A_MG_CC, 
+                    A_MN_CC = dt$A_MN_CC, 
+                    A_ZN_CC = dt$A_ZN_CC, 
+                    A_CU_CC = dt$A_CU_CC,
+                    A_C_BCS = dt$A_C_BCS, 
+                    A_CC_BCS = dt$A_CC_BCS,
+                    A_GS_BCS = dt$A_GS_BCS,
+                    A_P_BCS = dt$A_P_BCS,
+                    A_RD_BCS = dt$A_RD_BCS,
+                    A_EW_BCS = dt$A_EW_BCS,
+                    A_SS_BCS = dt$A_SS_BCS,
+                    A_RT_BCS = dt$A_RT_BCS,
+                    A_SC_BCS = dt$A_SC_BCS,
+                    M_COMPOST = dt$M_COMPOST,
+                    M_GREEN = dt$M_GREEN, 
+                    M_NONBARE = dt$M_NONBARE, 
+                    M_EARLYCROP = dt$M_EARLYCROP, 
+                    M_SLEEPHOSE = dt$M_SLEEPHOSE,
+                    M_DRAIN = dt$M_DRAIN,
+                    M_DITCH = dt$M_DITCH,
+                    M_UNDERSEED = dt$M_UNDERSEED,
+                    M_LIME = dt$M_LIME,
+                    M_NONINVTILL = dt$M_NONINVTILL, 
+                    M_SSPM = dt$M_SSPM, 
+                    M_SOLIDMANURE = dt$M_SOLIDMANURE,
+                    M_STRAWRESIDUE = dt$M_STRAWRESIDUE,
+                    M_MECHWEEDS = dt$M_MECHWEEDS,
+                    M_PESTICIDES_DST = dt$M_PESTICIDES_DST,
+                    ID = dt$ID,
+                    output = output
+                  )
+  
   
   # return output
   return(out)
   
 }
+
+#' Calculate the Open Bodem Index score for a series of fields belonging to a farm
+#' 
+#' This functions wraps the functions of the OBIC into one main function to calculate the score for Open Bodem Index (OBI).
+#' In contrast to obic_field, this wrapper uses a data.table as input.
+#' 
+#' @param dt (data.table) A data.table containing the data of the fields to calculate the OBI
+#' @param th_obi_c (numeric) A vector with the threshold values for farm evaluation of the chemical soil functions, values between 0 and 1, and in increasing order. 
+#' @param th_obi_p (numeric) A vector with the threshold values for farm evaluation of the physical soil functions, values between 0 and 1, and in increasing order. 
+#' @param th_obi_b (numeric) A vector with the threshold values for farm evaluation of the biological soil functions, values between 0 and 1, and in increasing order. 
+#' @param th_obi_e (numeric) A vector with the threshold values for farm evaluation of the environmental soil functions, values between 0 and 1, and in increasing order. 
```

Comment:
Do the thresholds have specific requirements for the length? Is it possible to use only 1 value, for example?

Reply:
no not necessarily. the length determines the classes that will be made. preferably it is fixed to three classes, but that is pragmatic choice. for the actual OBI-application i would not add more classes.